api/services/authentication/interface.js

```diff
@@ -0,0 +1,17 @@
+class AuthStrategyInterface {
+  async authenticate(credentials) {
+    throw new Error('Not Implemented');
+  }
+
+  constructor(app) {
+    this._app = app;
+  }
+
+  _generateJwt(user) {
+    return this._app.encodeJwt({
```

Comment:
Since this is an interface we'll just raise the NotImplementedExceptions.

Reply:
It is a common method used by all implementations. It would be reduntant code to write it again and again for each implementation separately.